### ISSUE TITLE
how to remove zero from the image

### ISSUE THREAD

**Issue by crebli** [gaospecial/ggVennDiagram#50]
hi, two much 0s were found in it, how to discard these digits from the venn image?

**Comment by gaospecial** [comment]
see: https://gaospecial.github.io/ggVennDiagram/articles/using-ggVennDiagram.html

**Comment by crebli** [comment]
hi，
Like this，how to neatly remove the useless 0 from the image?













在 2023-03-27 09:10:15，"Chun-Hui Gao" ***@***.***> 写道：

see: https://gaospecial.github.io/ggVennDiagram/articles/using-ggVennDiagram.html

**Comment by gaospecial** [comment]
我看不到图片。

**Comment by crebli** [comment]
![image](https://user-images.githubusercontent.com/63771650/227918252-0bd6700f-9653-41c1-af0c-bfbc7083fef8.png)

**Comment by crebli** [comment]
Like this. Many useless 0s make the image untidy. It's usefult to remove all of these 0 by a certain setting in the package.

**Comment by gaospecial** [comment]
> Like this. Many useless 0s make the image untidy. It's usefult to remove all of these 0 by a certain setting in the package.

see this: https://gaospecial.github.io/ggVennDiagram/articles/fully-customed.html

**Comment by crebli** [comment]
Nice！

**Comment by crebli** [comment]
Sorry! A small error with your newly added code:
ggplot() +
  # 1. region count layer
  geom_sf(aes(fill = count), data = venn_region(data)) +
  # 2. set edge layer
  geom_sf(aes(color = id), data = venn_setedge(data), show.legend = FALSE) +
  # 3. set label layer
  geom_sf_text(aes(label = count), fontface = "bold", data = venn_setlabel(data) %>% filter(count!= 0)) +
  # 4. region label layer
  geom_sf_label(aes(label = count), data = venn_region(data)) +
  theme_void()

Error in `stopifnot()`:
i In argument: `count != 0`.
Caused by error in `count != 0`:
! comparison (2) is possible only for atomic and list types

rlang::last_error()
<error/rlang_error>
Error in `stopifnot()`:
i In argument: `count != 0`.
Caused by error in `count != 0`:
! comparison (2) is possible only for atomic and list types
---
Backtrace:
  1. ggplot2::geom_sf_text(...)
 11. dplyr:::filter.data.frame(., count != 0)
 12. dplyr:::filter_rows(.data, dots, by)
 13. dplyr:::filter_eval(dots, mask = mask, error_call = error_call)
 15. mask$eval_all_filter(dots, env_filter)
 16. dplyr (local) eval()

**Comment by gaospecial** [comment]
Thanks for your feedback.

Corrected a mistake.

```r
genes <- paste0("gene",1:1000)
set.seed(20210302)
gene_list <- list(A = sample(genes,100),
                  B = sample(genes,200),
                  C = sample(genes,300),
                  D = sample(genes,200))

library(ggVennDiagram)
library(ggplot2)
library(dplyr)
venn <- Venn(gene_list)
data <- process_data(venn)
ggplot() +
  # 1. region count layer
  geom_sf(aes(fill = count), data = venn_region(data)) +
  # 2. set edge layer
  geom_sf(aes(color = id), data = venn_setedge(data), show.legend = FALSE) +
  # 3. set label layer
  geom_sf_text(aes(label = name), data = venn_setlabel(data)) +
  # 4. region label layer
  geom_sf_label(aes(label = count), data = venn_region(data) %>% filter(count > 10)) +
  theme_void()
```

![image](https://user-images.githubusercontent.com/107597/230257468-27137248-d2ba-469c-b89b-2758d106a3ba.png)

**Comment by crebli** [comment]
Nice. Thank you!
















At 2023-04-06 10:31:41, "Chun-Hui Gao" ***@***.***> wrote:

Thanks for your feedback.

Corrected a mistake.

genes<- paste0("gene",1:1000)
set.seed(20210302)
gene_list<-list(A= sample(genes,100),
                  B= sample(genes,200),
                  C= sample(genes,300),
                  D= sample(genes,200))

library(ggVennDiagram)
library(ggplot2)
library(dplyr)
venn<- Venn(gene_list)
data<- process_data(venn)
ggplot() +# 1. region count layer
  geom_sf(aes(fill=count), data= venn_region(data)) +# 2. set edge layer
  geom_sf(aes(color=id), data= venn_setedge(data), show.legend=FALSE) +# 3. set label layer
  geom_sf_text(aes(label=name), data= venn_setlabel(data)) +# 4. region label layer
  geom_sf_label(aes(label=count), data= venn_region(data) %>% filter(count>10)) +
  theme_void()

**Comment by crebli** [comment]
Hi,
A question with the ggVennDiagram is that all empty cells can be regarded as one cell in the column with less rows. For example, A: 60 items, B: 50 items, C: 40 items. We'll get A: 60, B:51, C:41 items after ploting as below. 
Please check it and find a resolution for this quetion during data input. 











在 2023-05-11 21:35:27，"Chun-Hui Gao" ***@***.***> 写道：

Closed #50 as completed.
